From 6e661937eaf4ada6e8679404083ca32da1a14d33 Mon Sep 17 00:00:00 2001
From: Arnd Bergmann <arnd@arndb.de>
Date: Thu, 8 Sep 2016 15:53:28 +0200
Subject: dax: use correct dev_t value
Git-commit: bc0a0fe94f33dd15edf2ed555bfc4d6dbb5e1995
Patch-mainline: v4.9-rc1
References: FATE#321135, FATE#321217, FATE#321256, FATE#321391, FATE#321393

The dev_t variable in devm_create_dax_dev() is used before it's
first set:

Drivers/dax/dax.c: In function 'devm_create_dax_dev':
drivers/dax/dax.c:205:39: error: 'dev_t' may be used uninitialized in this function [-Werror=maybe-uninitialized]
  inode = iget5_locked(dax_superblock, hash_32(devt + DAXFS_MAGIC, 31),
                                       ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
drivers/dax/dax.c:688:8: note: 'dev_t' was declared here

This reorders the code to how it looks correct to me.

Signed-off-by: Arnd Bergmann <arnd@arndb.de>
Fixes: 3bc52c45bac2 ("dax: define a unified inode/address_space for device-dax mappings")
Signed-off-by: Dan Williams <dan.j.williams@intel.com>
Signed-off-by: Johannes Thumshirn <jthumshirn@suse.de>

---
 drivers/dax/dax.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/dax/dax.c b/drivers/dax/dax.c
index 3f92bb2..e4398f4 100644
--- a/drivers/dax/dax.c
+++ b/drivers/dax/dax.c
@@ -598,6 +598,8 @@ struct dax_dev *devm_create_dax_dev(struct dax_region *dax_region,
 		goto err_minor;
 	}
 
+	dev_t = MKDEV(MAJOR(dax_devt), minor);
+	dev = &dax_dev->dev;
 	dax_dev->inode = dax_inode_get(&dax_dev->cdev, dev_t);
 	if (!dax_dev->inode) {
 		rc = -ENOMEM;
@@ -605,8 +607,6 @@ struct dax_dev *devm_create_dax_dev(struct dax_region *dax_region,
 	}
 
 	/* device_initialize() so cdev can reference kobj parent */
-	dev_t = MKDEV(MAJOR(dax_devt), minor);
-	dev = &dax_dev->dev;
 	device_initialize(dev);
 
 	cdev = &dax_dev->cdev;
-- 
1.8.5.6

